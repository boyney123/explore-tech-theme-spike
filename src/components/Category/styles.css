.category {
  box-shadow: 0 2px 3px rgba(10, 10, 10, 0.1), 0 0 0 1px rgba(10, 10, 10, 0.1);
  padding: 30px;
  display: flex;
  flex-direction: column;
  width: 100%;
  /* min-height: 300px; */

  /* display: flex; */
  /* justify-content: center; */
}

.category:hover {
  box-shadow: 0 8px 24px rgba(0, 0, 0, 0.2);
  cursor: pointer;
}

.category-container.column {
  display: flex;
}

.category img {
  width: 175px;
  margin: 0;
  padding: 0;
  margin: 0 auto;
}

.content .category h2 {
  margin-bottom: 10px;
}

.category .media-content .subtitle {
  line-height: 1.75;
  text-align: center;
}
