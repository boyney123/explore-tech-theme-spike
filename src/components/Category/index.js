import React from 'react'
import { navigate } from 'gatsby'
import iconToCategoryMap from '../../utils/icon-to-category'
import genericIcon from '../../icons/circle.svg'
import './styles.css'
import { siteMetadata } from '../../../gatsby-config'

const Category = ({ name } = {}) => {
  const { subtitle } = siteMetadata.categories[name]
  const icon = iconToCategoryMap[name] || genericIcon

  return (
    <div className="category-container column is-one-third">
      <div className="category" onClick={() => navigate(name)}>
        <img src={icon} alt="icon" />
        <h2 className=" has-text-centered">{name}</h2>

        <div className="category-content">
          <div className="media">
            <div className="media-content">
              <p className="subtitle is-6">{subtitle}</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

export default Category
