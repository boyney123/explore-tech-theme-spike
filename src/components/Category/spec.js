import React from 'react'
import { shallow } from 'enzyme'
import { navigate } from 'gatsby'
import { siteMetadata } from '../../../gatsby-config'

import Category from './'

const { categories } = siteMetadata

describe('Category', () => {
  for (const category in categories) {
    it(`when the name prop is set to "${category} is reads the siteConfig and uses the configured icon and subtitle`, () => {
      const wrapper = shallow(<Category name={category} />)
      expect(wrapper).toMatchSnapshot()
    })
  }

  it('when clicking on category the user is navigated to the name of the category', () => {
    const wrapper = shallow(<Category name="Components" />)
    wrapper.find('.category').simulate('click')
    expect(navigate).toHaveBeenCalledWith('Components')
  })
})
