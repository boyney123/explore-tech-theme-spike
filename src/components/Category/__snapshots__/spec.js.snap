// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category when the name prop is set to "Boilerplates is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Boilerplates
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            Get setup within minutes
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Charts is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Charts
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            Visualise data with React
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Components is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Components
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            List of React components
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Data is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Data
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            Managing data with React
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Developer is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Developer
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            React developer tools & experience
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Examples is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Examples
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            Explore example applications
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Forms is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Forms
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            Hands on with Forms
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Frameworks is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Frameworks
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            React UI Component Frameworks
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Integration is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Integration
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            React integration with others
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Libraries is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Libraries
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            Integrate with great libraries
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Redux is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Redux
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            All things Redux
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Routing is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Routing
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            Routing with React
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Styles is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Styles
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            Making your project look great
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Testing is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Testing
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            List of tools to help testing
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Tools is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Tools
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            Level up your React with tools
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Category when the name prop is set to "Tutorials is reads the siteConfig and uses the configured icon and subtitle 1`] = `
<div
  className="category-container column is-one-third"
>
  <div
    className="category"
    onClick={[Function]}
  >
    <img
      alt="icon"
      src="test-file-stub"
    />
    <h2
      className=" has-text-centered"
    >
      Tutorials
    </h2>
    <div
      className="category-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <p
            className="subtitle is-6"
          >
            Learn all things React
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
